datadogclient/datadog_client_test.go

```diff
@@ -175,623 +153,237 @@ var _ = Describe("DatadogClient", func() {
+			},
+		}))
+		Expect(metric.Tags).To(Equal(defaultTags))
 	})
 
-	It("sends a value 0 for the slowConsumerAlert metric when consumer error is not set", func() {
-		err := c.PostMetrics()
+	It("posts CounterEvents in JSON format & adds the metric prefix", func() {
+		k, v := makeFakeMetric("counterName", 1, 5, events.Envelope_CounterEvent, defaultTags)
+		metricsMap.Add(k, v)
+		k, v = makeFakeMetric("counterName", 2, 11, events.Envelope_CounterEvent, defaultTags)
+		metricsMap.Add(k, v)
+
+		err := c.PostMetrics(metricsMap)
 		Expect(err).ToNot(HaveOccurred())
 
 		Eventually(bodies).Should(HaveLen(1))
+
 		var payload datadogclient.Payload
 		err = json.Unmarshal(bodies[0], &payload)
 		Expect(err).NotTo(HaveOccurred())
-		Expect(payload.Series).To(HaveLen(3))
+		Expect(payload.Series).To(HaveLen(1))
 
-		errMetric := findSlowConsumerMetric(payload)
-		Expect(errMetric).NotTo(BeNil())
-		Expect(errMetric.Type).To(Equal("gauge"))
-		Expect(errMetric.Points).To(HaveLen(1))
-		Expect(errMetric.Points[0].Value).To(BeEquivalentTo(0))
+		metric := payload.Series[0]
+		Expect(metric.Type).To(Equal("gauge"))
+		Expect(metric.Metric).To(Equal("datadog.nozzle.counterName"))
+		Expect(metric.Points).To(Equal([]metrics.Point{
+			metrics.Point{
+				Timestamp: 1,
+				Value:     5.0,
+			},
+			metrics.Point{
+				Timestamp: 2,
+				Value:     11.0,
+			},
+		}))
+		Expect(metric.Tags).To(Equal(defaultTags))
 	})
 
-	It("unsets the slow consumer error once it publishes the alert to datadog", func() {
-		c.AlertSlowConsumerError()
+	It("breaks up a message that exceeds the FlushMaxBytes", func() {
+		for i := 0; i < 1000; i++ {
+			k, v := makeFakeMetric("metricName", 1000, uint64(i), events.Envelope_ValueMetric, defaultTags)
+			metricsMap.Add(k, v)
+		}
 
-		err := c.PostMetrics()
+		err := c.PostMetrics(metricsMap)
 		Expect(err).ToNot(HaveOccurred())
 
-		Eventually(bodies).Should(HaveLen(1))
-		var payload datadogclient.Payload
-		err = json.Unmarshal(bodies[0], &payload)
-		Expect(err).NotTo(HaveOccurred())
+		f := func() int {
+			return len(bodies)
+		}
 
-		errMetric := findSlowConsumerMetric(payload)
-		Expect(errMetric).NotTo(BeNil())
-		Expect(errMetric.Points[0].Value).To(BeEquivalentTo(1))
+		Eventually(f).Should(BeNumerically(">", 1))
+	})
+
+	It("discards metrics that exceed that max size", func() {
+		name := proto.String(strings.Repeat("some-big-name", 1000))
+		k, v := makeFakeMetric(*name, 1000, 5, events.Envelope_ValueMetric, defaultTags)
+		metricsMap.Add(k, v)
 
-		err = c.PostMetrics()
+		err := c.PostMetrics(metricsMap)
 		Expect(err).ToNot(HaveOccurred())
 
-		Eventually(bodies).Should(HaveLen(2))
-		err = json.Unmarshal(bodies[1], &payload)
-		Expect(err).NotTo(HaveOccurred())
+		f := func() int {
+			return len(bodies)
+		}
 
-		errMetric = findSlowConsumerMetric(payload)
-		Expect(findSlowConsumerMetric(payload)).ToNot(BeNil())
-		Expect(errMetric.Points[0].Value).To(BeEquivalentTo(0))
+		Consistently(f).Should(Equal(0))
```

Comment:
Why did you change Eventually to Consistently?

Reply:
I actually didn't change this, github's way of lining up the changes that were made just makes it look like I did.
Here's the original: https://github.com/DataDog/datadog-firehose-nozzle/blob/master/datadogclient/datadog_client_test.go#L547
And here's the new one: https://github.com/DataDog/datadog-firehose-nozzle/blob/2660d835d1b7d5fd7e21ba094ea9b2fc8c214a28/datadogclient/datadog_client_test.go#L361

Reply:
Okay that's fine!